### ISSUE TITLE
Add `Collections` support.

### ISSUE THREAD

**Issue by epage** [cobalt-org/cobalt.rs#368]
Use cases:
- A site might track talks separately from blog posts (see [johannhof's site](https://github.com/johannhof/johannhof.github.io)).  
- A collection of quotes
- Anything similar to a scrapbook

Proposal: support the user defining something like "posts" called a "collection".

### Jekyll

See [docs](https://jekyllrb.com/docs/collections/)

**Comment by epage** [comment]
Some ideas on refactoring to this:

This isn't about collections but more of notes on how I want to get there

Create Assets
- SassOptions -> SassBuilder in a sass.rs file

Collection
- Add option to Files to limit search to a specific sub-dir so we can walk each collection separately
- PostBuilder is a specialized builder for CollectionBuilder

Work already done:
- #315 was a small step
- #346 is also a step in this direction.

See also #312

**Comment by epage** [comment]
See also #309 for alternative ideas for how to do rss for collections

**Comment by epage** [comment]
One idea, spawned by #355, is for a folder to have a `_collection.yml` file.  Anything in that folder or deeper would then be a part of that collection.  

Pros
- It keeps the configuration close by.
- It makes it easy to detect the collection on a single file

Cons
- More "turds" dropped in the user's site
- We effectively have to walk the entire site (up until we find a `_collection.yml`) to discover all the collections
- We can't walk each collection in parallel.  Instead we walk the files and spit them out into each collection.

**Comment by mnivoliez** [comment]
Why not making a conf file with all collections and folder for each one?
like 
```yml
collections:
  - posts
  - talks
```
And in dir structure, we got folder posts (like usual) and talks.
What do you think?

**Comment by epage** [comment]
My current is to have all of the collections defined in `_cobalt.yml` in a manner [similar to posts](https://github.com/cobalt-org/cobalt.rs/blob/master/src/cobalt_model/collection.rs#L24).

```yml
collections:
- title: Cats
  rss: /cats.xml
  default:
    layout: cats.liquid
```

My comment above was more of a one off idea that I was exploring but probably won't do.

As mentioned above, the main question is how to handle RSS (see #309) but I think this should get implemented and worry about making RSS nicer, later.

**Comment by Geobert** [comment]
Do we have though on pagination?

**Comment by Geobert** [comment]
We can mix both ideas, using epage format and the title as the folder name so we can walk collections in parallel. This whole collection concept looks like gutenberg's `Section` as well (a folder = a section)

**Comment by epage** [comment]
Yeah, I've been looking at gutenburg and others when designing this.  You can see the where I'm currently looking at going with collections by looking at the [`CollectionBuilder`](https://github.com/cobalt-org/cobalt.rs/blob/master/src/cobalt_model/collection.rs#L26).  We'd make a `CollectionConfig` similar to `PostConfig` and `PageConfig` that basically exposes every `CollectionBuilder feature in the config format.

**Comment by dmoles** [comment]
What's the status of this feature? The existence of `collections.posts` suggests the possibility of collections other than `posts`, but there doesn't seem to be any obvious way to create them (e.g. the syntax [mentioned above](https://github.com/cobalt-org/cobalt.rs/issues/368#issuecomment-360126968) produces `Error: failed to parse config` with ``unknown field `collections`​``).
